test/cluster/cluster-test.sh

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+set -euxo pipefail
+
+# setup DIR for easier pathing /Users/dax/work/sourcegraph/test/cluster
+DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)""
+# cd to repo root
+cd "$(dirname "${BASH_SOURCE[0]}")/../.."
+
+function cluster_setup() {
+  git clone --depth 1 \
+    https://github.com/sourcegraph/deploy-sourcegraph.git \
+    "$DIR/deploy-sourcegraph"
+
+  gcloud container clusters get-credentials default-buildkite --zone=us-central1-c --project=sourcegraph-ci
+
+  export NAMESPACE="cluster-ci-$BUILDKITE_BUILD_NUMBER"
+  kubectl create ns "$NAMESPACE" -oyaml --dry-run | kubectl apply -f -
+  kubectl apply -f "$DIR/storageClass.yaml"
+  kubectl config set-context --current --namespace="$NAMESPACE"
+  kubectl config current-context
+  sleep 15 #wait for namespace to come up
+  kubectl get -n "$NAMESPACE" pods
+
+  pushd "$DIR/deploy-sourcegraph/"
+  pwd
+  # script contains relative paths :(
+  ./create-new-cluster.sh
```

Comment:
```suggestion
  ./tools/update-docker-tags.sh $CANDIDATE_VERSION
  ./create-new-cluster.sh
```

I think we need something like this to ensure builds actually test the commit being built. However:

* these candidate images are on `us.gcr.io/sourcegraph-dev` so this won't work out of the box :/
* you need gcloud auth setup to pull these candidate images, not sure if that will work without additional setup

Reply:
Worked with @bobheadxi offline. The `update-docker-tags` tool currently doesn't have this functionality and it will need to be added.

Reply:
@daxmc99 seems to have found a much cleaner way around this :)

Reply:
see https://github.com/sourcegraph/sourcegraph/pull/15678#discussion_r525877821